docs/introduction/faq.md

```diff
@@ -0,0 +1,109 @@
+- the product must not be set as hidden
+- if the attribute "selling start date" is filled in, the value of this attribute must be set to the date in the past
+- if the attribute "selling end date" is filled in, the value of this attribute must be set to the date in the future
+- the product must have a name for the specific locale
+- if the product is a variant, there must exist calculated price for this variant for the specific pricing group
+- if the product is a variant, her main variant must not be set as hidden
+- if the product is the main variant, at least one of her variants must be visible
+
+**Offered** - products that satisfied the conditions for **visible** and at the same time they appear in the database with the attribute `calculatedSellingDenied` = `FALSE`.
+The `calculatedSellingDenied` attribute shows whether the product is already sold out or if the product is a variant with the main variant that is not set up with selling denied.
+
+**Listable** - products that satisfied the conditions for **offered** and at the same time, these products are not the variants.
+By default, the specific variants are not included in the product lists.
+Only the main variants are included in the product lists.
+
+**Sellable** - products that satisfied the conditions for **offered** and at the same time, these products are not the main variants (in other words, these products are either the specific variants or they are the standard products).
+Sellable products are products that can actually be purchased.
+Only the standard products or the specific variants can actually be purchased.
+
+## How calculated attributes work?
+Some attributes that are used on the Shopsys Framework are not set directly, but their value stands out as a result of the values of other attributes.
+The values of these special attributes are calculated automatically.
+For example, if a category of products does not have a name for a locale of the specific domain, this category will be automatically set as invisible on this domain.
+The recalculations of these special attributes can be initialized as `immediate` or `scheduled`:
+
+**immediate** - recalculation is initialized when the event `kernel.response` is caught.
+See a class `Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceRecalculator` and a method `onKernelResponse`.
+
+**scheduled** - recalculation is initialized later.
+A product or a category of products can be marked for scheduled recalculation, the recalculation itself is initialized with a cron module, see a class `Shopsys\FrameworkBundle\Command\RecalculationsCommand`.
+
+For example, a method `edit` of a class `Shopsys\FrameworkBundle\Model\Product\ProductFacade` calls a method `scheduleProductForImmediateRecalculation` of a class `Shopsys\FrameworkBundle\Model\Product\Availability\ProductAvailabilityRecalculationScheduler`.
+Through this request, the product is marked for `immediate` recalculation of availability.
```

Comment:
The product is marked for `immediate` recalculation of availability by this request.

Reply:
Done